### PR TITLE
Ensure fatal errors stop distributors

### PR DESCRIPTION
**Pull request by rdooley**
* Shopify/tempo-shopify#221

Want to see the ci run on this

### CODE REVIEW

**Comment by rdooley** [comment]
I'm gonna go do a diff commit on my fork to get this stuff upstream

**Comment by rdooley** [comment]
deleting the subscription does trigger a fatal
```
2022-11-10T13:47:06.270-0800    error   components/host_wrapper.go:40   Component fatal error   {"kind": "receiver", "name": "googlecloudpubsublite", "pipeline": "traces", "error": "assigner(projects/shopify-tracing-testing/locations/us-east1-c/subscriptions/rees-test2): rpc error: code = NotFound desc = Resource projects/30220954307/locations/us-east1-c/subscriptions/rees-test2 of type SUBSCRIPTION does not exist."}
go.opentelemetry.io/collector/service/internal/components.(*hostWrapper).ReportFatalError
        /Users/rdooley/pkg/mod/go.opentelemetry.io/collector@v0.61.0/service/internal/components/host_wrapper.go:40
github.com/Shopify/opentelemetry-collector-shopify/receiver/googlecloudpubsublitereceiver.(*pubsubliteReceiver).Start.func2
        /Users/rdooley/src/github.com/Shopify/opentelemetry-collector-shopify/receiver/googlecloudpubsublitereceiver/receiver.go:242
2022-11-10T13:47:06.270-0800    error   service/collector.go:190        Asynchronous error received, terminating process        {"error": "assigner(projects/shopify-tracing-testing/locations/us-east1-c/subscriptions/rees-test2): rpc error: code = NotFound desc = Resource projects/30220954307/locations/us-east1-c/subscriptions/rees-test2 of type SUBSCRIPTION does not exist."}
go.opentelemetry.io/collector/service.(*Collector).Run
```

**Comment by rdooley** [comment]
need to retest on monday with the ctx.done bit in there

**Review by gravelg — APPROVED** [review]
Looks good, not sure what is going on with those failing tests however

File: Makefile [hunk 1]
```diff
@@ -202,16 +202,6 @@ update-mod:
 ### Release (intended to be used in the .github/workflows/release.yml)
 # Note: goreleaser is pinned to 1.6.3 because newer versions require go1.18. Once we have migrated 
 #  the project and CI pipeline to 1.18 we can go back to using @latest.
-$(GORELEASER):
```
[the remark this answers is not in the document]
Comment (rdooley) [reply Makefile:205]: ah whoops this was so i could run make locally

**Pull request merged**